app/src/main/java/com/github/geohunt/app/ui/components/profile/ProfilePageViewModel.kt

```diff
@@ -84,6 +90,18 @@ open class ProfilePageViewModel(
     private fun fetch() {
         viewModelScope.launch {
             try {
+                if (!isSelf) {
+                    when (profileVisibilityRepository.getProfileVisibility(uid).first()) {
+                        ProfileVisibility.PUBLIC -> {}
+                        ProfileVisibility.PRIVATE -> { _isPrivate.value = true; return@launch }
+                        ProfileVisibility.FOLLOWING_ONLY -> {
+                            val isPrivate = !followRepository.doesFollow(uid, authedUid).first()
+                            _isPrivate.value = isPrivate
+                            if (isPrivate) return@launch
+                        }
+                    }
+                }
```

Comment:
**Note** (for future reference): We should introduce rules within firebase to enforce this mechanism

Reply:
It is just for future reference don't change anything

Reply:
> Note (for future reference): We should introduce rules within firebase to enforce this mechanism

Back-end ? 🙃